packages/fx-core/src/component/generator/templates/templateGenerator.ts

```diff
@@ -73,35 +46,48 @@ export class DefaultTemplateGenerator implements TemplateGenerator {
     }),
   ])
   public async run(
```

Comment:
The entry point for overwrite methods is the `run()` method. Therefore, would it be feasible to consolidate parameters such as context, inputs, and actionContext during the execution of the `run(`) method? This could ensure a uniform context.

Reply:
I didn't get what you mean. Could you provide an example?

Reply:
Could we possibly consider writing it as: 
`run(
    context: Context,
    inputs: Inputs,
    destinationPath: string,
    actionContext?: ActionContext
){
    this.ctx = context;
    this.inputs = inputs;
}`

Reply:
TTK does not block users from creating two projects at the same time. Making the generator stateful may cause unexpected issues.